src/main/java/de/mediathekview/mserver/crawler/hr/tasks/HrSendungsfolgedetailsTask.java

```diff
@@ -2,58 +2,199 @@
+      Map<Resolution, String> videoUrls, Optional<String> untertitelUrlText, Duration dauer, Optional<LocalDateTime> time) {
+    try {
+      final Film newFilm = new Film(UUID.randomUUID(), crawler.getSender(), title,
+          topic, time.orElse(LocalDate.now().atStartOfDay()), dauer);
+      newFilm.setWebsite(new URL(aUrlDto.getUrl()));
+      if (beschreibung.isPresent()) {
+        newFilm.setBeschreibung(beschreibung.get());
+      }
+
+      for (Entry<Resolution, String> videoUrl : videoUrls.entrySet()) {
+        newFilm.addUrl(videoUrl.getKey(),
+            CrawlerTool.uriToFilmUrl(new URL(videoUrl.getValue())));
+      }
+
+      if (untertitelUrlText.isPresent()) {
+        newFilm.addSubtitle(new URL(untertitelUrlText.get()));
+      }
+
+      final Optional<FilmUrl> defaultUrl = newFilm.getDefaultUrl();
+      if (defaultUrl.isPresent()) {
+        newFilm.setGeoLocations(CrawlerTool.getGeoLocations(crawler.getSender(),
+            defaultUrl.get().getUrl().toString()));
+      }
+
+      taskResults.add(newFilm);
+      crawler.incrementAndGetActualCount();
+      crawler.updateProgress();
+    } catch (final MalformedURLException malformedUrlException) {
+      LOG.fatal("A HR URL can't be parsed.", malformedUrlException);
+      crawler.printErrorMessage();
+      crawler.incrementAndGetErrorCount();
+      crawler.updateProgress();
+    }
+  }
+
+  private boolean checkPreConditions(Optional<String>[] topicAndTitle, Map<Resolution, String> videoUrls,
+      Optional<Duration> dauer, final String aUrl) {
+    if (!topicAndTitle[1].isPresent()) {
+      crawler.printMissingElementErrorMessage(aUrl + ": Titel");
+      return false;
+    }
+
+    if (!topicAndTitle[0].isPresent()) {
+      crawler.printMissingElementErrorMessage(aUrl + ": Thema");
+      return false;
+    }
+
+    if (videoUrls.isEmpty()) {
+      crawler.printMissingElementErrorMessage(aUrl + ": Video-Url");
+      return false;
+    }
+
+    if (!dauer.isPresent()) {
+      crawler.printMissingElementErrorMessage(aUrl + ": Dauer");
+      return false;
+    }
+
+    return true;
+  }
+
+  private static Optional<String>[] getTopicAndTitle(final Document aDocument) {
+    Optional<String>[] topicAndTitle = new Optional[2];
+
+    // Komplizierte Logik, da der HR nicht eindeutig das Thema und den Titel setzt
+    Elements breadcrumbElements = aDocument.select(THEMA_SELECTOR1);
+    if (breadcrumbElements.size() > 2) {
+      // Bei den meisten Seiten lässt sich das Thema aus der Breadcrumb auslesen
+      // Der Titel ist dann mal die Überschrift mal die Unterüberschrift
+      topicAndTitle[0] = Optional.of(breadcrumbElements.get(2).text());
+      final Optional<String> title = HtmlDocumentUtils.getElementString(THEMA_SELECTOR2, aDocument);
+      if (title.isPresent() && topicAndTitle[0].isPresent() && topicAndTitle[0].get().compareToIgnoreCase(title.get()) == 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)